test/unit/adapters/rpc-server.spec.ts

```diff
@@ -68,12 +78,54 @@ describe('createRpcServerComponent', () => {
 
   describe('attachUser', () => {
     it('should attach a user and register transport events', () => {
-      const mockTransport = { on: jest.fn() } as unknown as Transport
       const address = '0x123'
 
       rpcServer.attachUser({ transport: mockTransport, address })
 
       expect(mockTransport.on).toHaveBeenCalledWith('close', expect.any(Function))
+      expect(attachTransportMock).toHaveBeenCalledWith(mockTransport, {
+        subscribers: expect.any(Object),
+        address
+      })
+    })
+
+    it('should clean up subscribers when transport closes', () => {
```

Comment:
can we add a expect about this call or the only way to testing is through `rpcServer.detachUser` current behavior?

Reply:
or add a test with its counterpart:

`should throw when detaching an user from RPC if subscriber was not priorly cleaned`